File: first_non_repeated_char/first_non_repeated_char.py
# First Non-Repeated Character
# http://www.ardendertat.com/2011/11/14/programming-interview-questions-15-first-non-repeated-character-in-string/

class Solution:
	def first_non_rep(self, s):
		count = {}
		for c in s:
			if c not in count:
				count[c] = 0
			count[c] += 1

		return [c for c in s if count[c] == 1]
		return "NONE"

s = Solution()
string = "AABBCDD"
print(''.join(s.first_non_rep(string)))

File: first_non_repeated_char/first_non_repeated_char.php
<?php

class Solution {

	// O(n)
	public function firstNonRepeatedChar($str) {
		$count = [];

		foreach(str_split($str) as $char) {
			if (!array_key_exists($char, $count)) {
				$count[$char] = 0;
			}

			$count[$char] += 1;
		}

		foreach($count as $key=>$value) {
			if ($value == 1) {
				return $key;
			}
		}

		return null;
	}

}

$sol = new Solution();
$str = "AABBCDD";
echo $sol->firstNonRepeatedChar($str);
